go/vt/vtgate/semantics/early_rewriter.go

```diff
@@ -91,6 +89,12 @@ func (r *earlyRewriter) up(cursor *sqlparser.Cursor) error {
 			r:    r,
 		}
 		return r.handleOrderBy(cursor.Parent(), iter)
+	case *sqlparser.Where:
+		if node.Type != sqlparser.HavingClause {
+			return nil
+		}
+
+		return r.handleHavingClause(node, cursor.Parent())
 	}
 	return nil
```

Comment:
nit:
```suggestion
	case *sqlparser.Where:
		if node.Type == sqlparser.HavingClause {
			return r.handleHavingClause(node, cursor.Parent())
		}
	}
	return nil
```

Reply:
I agree, but I'm trying to minimize the refactorings, since we want to backport this